pyrax/cloudcdn.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c)2013 Rackspace US, Inc.
+
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from functools import wraps
+import re
+
+from pyrax.client import BaseClient
+import pyrax.exceptions as exc
+from pyrax.manager import BaseManager
+from pyrax.resource import BaseResource
+import pyrax.utils as utils
+
+
+class CloudCDNFlavor(BaseResource):
+    pass
+
+class CloudCDNFlavorManager(BaseManager):
+
+    def list(self):
+        resp, resp_body = self.api.method_get("/%s" % self.uri_base)
+        return [CloudCDNFlavor(self, info)
+                for info in resp_body[self.plural_response_key]]
+
+    def get(self, flavor_id):
+        resp, resp_body = self.api.method_get(
+                "/%s/%s" % (self.uri_base, flavor_id))
+        return CloudCDNFlavor(self, resp_body)
+
+
+class CloudCDNService(BaseResource):
+
+    def update(self, changes):
+        self.manager.update(self.id, changes)
+
+    def delete(self):
+        self.manager.delete(self)
+
+    def delete_assets(self, url=None, all=False):
+        self.manager.delete_assets(self.id, url, all)
+
+
+class CloudCDNServiceManager(BaseManager):
+
+    def create(self, name, flavor_id, domains, origins,
+               restrictions=None, caching=None):
+
+        body = {"name": name,
+                "flavor_id": flavor_id,
+                "domains": domains,
+                "origins": origins,
+                "restrictions": restrictions or [],
+                "caching": caching or []}
+        resp, resp_body = self.api.method_post("/%s" % self.uri_base,
+                                               body=body)
+
+        return resp.headers.get("location")
+
+    def update(self, service_id, changes):
```

Comment:
All other SDKs take the new service object, compute the JSON Patch by diff'ing it against the current service object, and then send the JSON Patch representation over the wire. This way the client of the SDK does not have the burden of computing the JSON Patch (i.e. the value of the `changes` argument). Could you do something similar here, perhaps using http://python-json-patch.readthedocs.org/en/latest/mod-jsonpatch.html#jsonpatch.make_patch?